src/device-registry/utils/generate-filter.js

```diff
@@ -62,65 +63,74 @@ const generateDeviceFilter = (
   channel,
   location,
   siteName,
-  mapAddress
+  mapAddress,
+  primary,
+  active
 ) => {
-  if (tenant && name && !channel && !location && !siteName && !mapAddress) {
+
+  if(!tenant){
+    return {}
+  }
+
+  let filter = {};
+
+  if(name){
     let regexExpression = generateRegexExpressionFromStringElement(name);
-    let filter = {
-      name: { $regex: regexExpression, $options: "i" },
-    };
-    return filter;
-  } else if (
-    tenant &&
-    !name &&
-    channel &&
-    !location &&
-    !siteName &&
-    !mapAddress
-  ) {
-    return {
-      channelID: channel,
-    };
-  } else if (
-    tenant &&
-    !name &&
-    !channel &&
-    location &&
-    !siteName &&
-    !mapAddress
-  ) {
-    return {
-      locationID: location,
-    };
-  } else if (
-    tenant &&
-    !name &&
-    !channel &&
-    !location &&
-    siteName &&
-    !mapAddress
-  ) {
+    filter['name'] = { $regex: regexExpression, $options: "i" };
+  }
+
+  if(channel){
+    filter['channelID'] = channel;
+  }
+
+  if(location){
+    filter['locationID'] = location;
+  }
+
+  if(siteName){
     let regexExpression = generateRegexExpressionFromStringElement(siteName);
-    let filter = {
-      siteName: { $regex: regexExpression, $options: "i" },
-    };
-    return filter;
-  } else if (
-    tenant &&
-    !name &&
-    !channel &&
-    !location &&
-    !siteName &&
-    mapAddress
-  ) {
+    filter['siteName'] = { $regex: regexExpression, $options: "i" };
+  }
+
+  if(mapAddress){
     let regexExpression = generateRegexExpressionFromStringElement(mapAddress);
-    let filter = {
-      locationName: { $regex: regexExpression, $options: "i" },
-    };
-    return filter;
-  } else if (tenant && !name && !channel && !location) {
-    return {};
+    filter['locationName'] = { $regex: regexExpression, $options: "i" };
+  }
+
+  if (primary) { 
+
+    const primaryStr = primary + "";
+    
+    if(primaryStr.toLowerCase() == 'yes'){
```

Comment:
@sserurich, the change was made following @Baalmart comments https://github.com/airqo-platform/AirQo-api/pull/384#discussion_r633295302

........ Using Boolean values could potentially bring about some runtime and/or logical errors during operation. " ....... 
"Because of the way we use "truthiness" when checking for presence of a value in JS, I suggest that we explore using values/strings like "yes" and "no" when checking the values for the query parameter. "